Packages/MIES/MIES_GuiPopupMenuExt.ipf

```diff
@@ -0,0 +1,623 @@
+			endfor
+			multithread itemListWave[] = menuWave[p]
+		endif
+	endif
+End
+
+/// @brief Sets the menu to show a non selectable "_none_" entry.
+static Function PEXT_SetDisabledMenu()
+
+	WAVE/T itemListWave = GetPopupExtMenuWave()
+	Redimension/N=1 itemListWave
+	itemListWave = ""
+	SetDimLabel ROWS, 0, $MENUNAME_UNUSED, itemListWave
+End
+
+/// @brief Automatically splits a 1D text wave with menu items to submenus
+///        This function does not name submenus
+///
+/// @param[in] menuList 1d text wave with menu items
+/// @param[in] method [optional, default = PEXT_SUBSPLIT_DEFAULT] sets how the menu items are split to sub menus @sa PEXT_SubMenuSplitting
+/// @returns 1d text wave where each element contains a list of menu items. Each element represents a sub menu.
+Function/WAVE PEXT_SplitToSubMenus(menuList[, method])
+	WAVE/T menuList
+	variable method
+
+	variable subMenuCnt, beginitem, endItem, i, j
+	variable numItems, remainItems, menuPos, subIndex, subMenuLength
+	string begEntry, endEntry, checkEntry
+
+	method = ParamIsDefault(method) ? PEXT_SUBSPLIT_DEFAULT : method
+
+	if(method == PEXT_SUBSPLIT_DEFAULT)
+		Sort/A menuList, menuList
+		subMenuCnt = trunc(DimSize(menuList, ROWS) / NUM_SUBENTRIES) + 1
+		subMenuCnt = subMenuCnt > MAX_SUBMENUS ? MAX_SUBMENUS : subMenuCnt
+
+		Make/FREE/T/N=(subMenuCnt) splittedMenu
+
+		for(i = 0; i < subMenuCnt; i++)
+			beginItem = i * NUM_SUBENTRIES
+			endItem = i == subMenuCnt - 1 ? DimSize(menuList, ROWS) - 1 : beginItem + NUM_SUBENTRIES - 1
+
+			for(j = beginitem; j < enditem; j++)
+				splittedMenu[i] = AddListItem(menuList[j], splittedMenu[i], ";", Inf)
+			endfor
+		endfor
+
+	elseif(method == PEXT_SUBSPLIT_ALPHA)
+		Sort/A menuList, menuList
+
+		numItems = DimSize(menuList, ROWS)
+		Make/FREE/T/N=(MAX_SUBMENUS) splittedMenu
+		do
+			remainItems = DimSize(menuList, ROWS) - menuPos
+			if(remainItems < NUM_SUBENTRIES || subIndex == MAX_SUBMENUS - 1)
+				subMenuLength = remainItems
+			else
+				begEntry = menuList[menuPos]
+				endEntry = menuList[menuPos + NUM_SUBENTRIES - 1]
+				if(!CmpStr(begEntry[0], endEntry[0]))
+					subMenuLength = NUM_SUBENTRIES
+				else
+					subMenuLength = NUM_SUBENTRIES - 1
+					do
+						subMenuLength--
+						checkEntry = menuList[menuPos + subMenuLength]
+					while(!CmpStr(endEntry[0], checkEntry[0]))
+					subMenuLength++
+				endif
+			endif
+
+			for(i = 0; i < subMenuLength; i++)
+				splittedMenu[subIndex] = AddListItem(menuList[menuPos + i], splittedMenu[subIndex], ";", Inf)
+			endfor
+
+			subIndex++;
+			menuPos += subMenuLength
+		while(menuPos < numItems)
+
+		Redimension/N=(subIndex) splittedMenu
+	else
+		ASSERT(0, "Unknown method for automatically splitting to submenus")
+	endif
+
+	return splittedMenu
+End
+
+/// @brief Automatically generates submenu names and applies them on the given wave
+///
+/// @param[in] splittedMenu 1d text wave with menu item lists for sub menus as returned by PEXT_SplitToSubMenus()
```

Comment:
@MichaelHuth `splittedMenu` -> `splitMenu`

Reply:
I renamed the WAVE reference to splitMenu in all related functions.